### PR TITLE
Apidoc: output API version instead of hardcoded product version

### PR DESCRIPTION
**Pull request by cbosdo**
## What does this PR change?

Changes the hardcoded product version in the apidocs by the API version.

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: doc tooling fix

- [X] **DONE**

## Test coverage
- No tests: doc tooling fix

- [X] **DONE**

## Links

Fixes SUSE/spacewalk#11828

- [X] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [X] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: java/code/internal/src/com/redhat/rhn/internal/doclet/ApiDoclet.java [hunk 1]
```diff
@@ -132,13 +142,14 @@ protected ApiDoclet() {
 
     /**
      * @return The documentation writer for the Doclet implementation
-     *
-     * @param outputIn the folder where the result needs to be written
+     *@param outputIn the folder where the result needs to be written
```
Comment (mcalmer) [inline java/code/internal/src/com/redhat/rhn/internal/doclet/ApiDoclet.java:145]: nitpick - isn't there a space missing?

Comment (cbosdo) [reply java/code/internal/src/com/redhat/rhn/internal/doclet/ApiDoclet.java:145]: indeed, looks ugly

**Comment by cbosdo** [comment]
> Maybe it's because I am not a Java guy, but where is `$apiVersion` defined now?

It is passed by the ant script and comes from the `java.apiversion` property in `java/conf/rhn_java.conf` file

File: java/code/internal/src/com/redhat/rhn/internal/doclet/ApiDoclet.java (hunk 1 above)
Comment (cbosdo) [reply java/code/internal/src/com/redhat/rhn/internal/doclet/ApiDoclet.java]: fixed now

**Comment by jcayouette** [comment]
Im not getting an api version with this change. Its dumping the variable in, but there is no replacement. Im not sure if this is intended or not. Of course we can manually add it before building. 

Asciidoc:
```
= SUSE Manager API Documentation
SUSE Manager
API version: $apiVersion
```

Docbook:
```
<book lang="en">
  <title>SUSE Manager API Documentation</title>
  <bookinfo>
    <pubdate><?dbtimestamp?></pubdate>
    <productname>SUSE Manager</productname>
    <productnumber>$apiVersion</productnumber>
```

**Pull request merged**